src/pam_google_authenticator.c

```diff
@@ -385,6 +392,38 @@ static int drop_privileges(pam_handle_t *pamh, const char *username, int uid,
   gid_t gid = pw->pw_gid;
   free(buf);
 
+  // to access the secret file, we may have to be member of the same groups as the 
```

Comment:
Break out all this grouplist fetching into its own function, so that it's easy to follow the failure path, that should presumably be to degrade to the current behaviour of no extra groups.

Bonus points for  hiding this behind a `HAVE_SETGROUPLIST` since it's apparently nonstandard.